src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/AbstractCloudFoundryDeployer.java

```diff
@@ -167,6 +172,12 @@ String getDockerImage(AppDeploymentRequest request) {
 	Path getApplication(AppDeploymentRequest request) {
 		try {
 			if (!request.getResource().getURI().toString().startsWith("docker:")) {
+				if (request.getResource() instanceof UrlResource) {
```

Comment:
I think we need to log a similar message in all the cases not just for the URLResource as the URL resource corresponds to `http` based only.

Reply:
After discussing with @dturanski, I learned that https://github.com/spring-cloud/spring-cloud-deployer/pull/303 addresses this in a generic way and hence this PR can be merged as is.